Presentation.Web/app/components/data-processing/tabs/data-processing-registration-sub-data-processor-modal.controller.ts

```diff
@@ -0,0 +1,189 @@
+                null,
+                false);
+        }
+
+        private bindTransferToThirdCountriesOptions(subDataProcessor: Models.DataProcessing.IDataProcessorDTO | null): Models.ViewModel.Generic.ISingleSelectionWithFixedOptionsViewModel<Models.Api.Shared.YesNoUndecidedOption> {
+            const thirdCountriesOptions = new Models.ViewModel.Shared.YesNoUndecidedOptions();
+
+            let selectedOption = null;
+            if (subDataProcessor) {
+                selectedOption = thirdCountriesOptions.getById(subDataProcessor.transferToInsecureThirdCountries);
+            }
+
+            return {
+                selectedElement: selectedOption,
+                select2Config: this.select2LoadingService.select2LocalDataNoSearch(() => thirdCountriesOptions.options, false),
+                elementSelected: (newElement) => {
+
+                    this.viewModel.updateTransferToInsecureThirdCountry(newElement);
+
+                    if (this.viewModel.transferToInsecureThirdCountryId !== Models.Api.Shared.YesNoUndecidedOption.Yes) {
+                        this.subDataProcessorInsecureThirdCountriesConfig.selectedElement = null;
+                    }
+                }
+            };
+        }
+
+        private bindInsecureThirdCountries(subDataProcessor: Models.DataProcessing.IDataProcessorDTO | null): Models.ViewModel.Generic.ISingleSelectionWithFixedOptionsViewModel<Models.Generic.NamedEntity.NamedEntityWithDescriptionAndExpirationStatusDTO> {
+            return Helpers.Select2MappingHelper.createNewNamedEntityWithDescriptionAndExpirationStatusDtoViewModel(subDataProcessor?.insecureCountry,
+                this.dataProcessingRegistrationOptions.thirdCountryOptions,
+                (newElement) => this.viewModel.updateInsecureThirdCountry(newElement),
+                this.select2LoadingService,
+                null,
+                false);
+        }
+            
+        private bindSubDataProcessor(): Models.ViewModel.Generic.ISingleSelectionWithFixedOptionsViewModel<Models.DataProcessing.IDataProcessorDTO> {
+
+            const pageSize = 100;
+            return {
+                selectedElement: null,
+                select2Config: this.select2LoadingService.loadSelect2WithDataSource(
+                    (query) => this
+                        .dataProcessingRegistrationService
+                        .getApplicableSubDataProcessors(this.dataProcessingRegistration.id, query, pageSize)
+                        .then(results => Helpers.Select2MappingHelper.mapDataProcessingSearchResults(results)),
+                    true),
+                elementSelected: (newElement) => {
+                    this.viewModel.updateSubDataProcessor(newElement);
+                }
+            };
+        }
+
+        private getSubDataProcessor(): Models.DataProcessing.IDataProcessorDTO | null {
+            return _.find(this.dataProcessingRegistration.subDataProcessors, { id: this.subDataProcessorId });
```

Comment:
this is only used during init() .. also it does not test for if the sdp id is not set.. isEdit could be used for it..

Reply:
or isEdit could be determined on the presence of the sdp id..

Reply:
not being null